src/inventory.rs

```diff
@@ -193,4 +193,162 @@ mod inventory_tests {
 
         assert_eq!(inv.classes, expected_classes);
     }
+
+    use crate::types::Value;
+    fn literal(v: &str) -> Value {
+        Value::Literal(v.to_string())
+    }
+
+    fn sequence(i: &[&str]) -> Value {
+        Value::Sequence(i.iter().map(|s| literal(s)).collect::<Vec<Value>>())
+    }
+
+    fn check_compose_node_name_shared(inv: &Inventory, invabsdir: &str) {
+        let mut nodes = inv.nodes.keys().collect::<Vec<_>>();
+        nodes.sort();
+        assert_eq!(
+            nodes,
+            vec!["a", "a.1", "b.1", "c.1", "c._c.1", "d", "d1", "d2"]
+        );
+
+        for n in nodes {
+            if n == "a.1" {
+                // not checking a.1 here, since it's the one node that is rendered differently
+                // between compat- and nocompat-mode.
+                continue;
+            }
+            let node = &inv.nodes[n];
+            assert_eq!(node.reclass.node, *n);
+            assert_eq!(node.reclass.name, *n);
+            let expected_full_name = node.parameters.get(&"node_name".into()).unwrap();
+            let expected_short_name = node.parameters.get(&"short_name".into()).unwrap();
+            let expected_path = node.parameters.get(&"path".into()).unwrap();
+            let expected_parts = node.parameters.get(&"parts".into()).unwrap();
+            let expected_uri_suffix = node.parameters.get(&"uri_suffix".into()).unwrap();
+            assert_eq!(
+                node.reclass.uri,
+                format!(
+                    "yaml_fs://{invabsdir}/nodes/{}",
+                    expected_uri_suffix.as_str().unwrap()
+                )
+            );
+            let params_reclass_name = node
+                .parameters
+                .get(&"_reclass_".into())
+                .unwrap()
+                .get(&"name".into())
+                .unwrap();
+            assert_eq!(
+                params_reclass_name.get(&"full".into()),
+                Some(expected_full_name)
+            );
+            assert_eq!(
+                params_reclass_name.get(&"short".into()),
+                Some(expected_short_name)
+            );
+            assert_eq!(params_reclass_name.get(&"path".into()), Some(expected_path));
+            assert_eq!(
+                params_reclass_name.get(&"parts".into()),
+                Some(expected_parts)
+            )
+        }
+    }
+
+    #[test]
```

Comment:
I admittedly don't understand why there's tests here as well as in the tests folder, but I assume it's meant to be this way.

Reply:
Most projects keep Rust unit tests in the source files in a module marked `#[cfg(test)]` so it's only compiled when running tests. `tests/` has a few Rust integration tests (but those could probably be dropped at this point), as well as the Python tests which exercise the Python interface exposed by the Rust code through PyO3.